routes/helpers/firewall.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class Clover
+  def firewall_list_dataset
+    @project.firewalls_dataset.authorized(current_account.id, "Firewall:view")
+  end
+
+  def firewall_list_api_response(dataset)
+    dataset = dataset.where(location: @location) if @location
+    result = dataset.eager(:firewall_rules).paginated_result(
+      start_after: request.params["start_after"],
+      page_size: request.params["page_size"],
+      order_column: request.params["order_column"]
+    )
+
+    {
+      items: Serializers::Firewall.serialize(result[:records]),
+      count: result[:count]
+    }
+  end
+
+  def firewall_post(firewall_name)
+    Authorization.authorize(current_account.id, "Firewall:create", @project.id)
+    Validation.validate_name(firewall_name)
+    @location ||= LocationNameConverter.to_internal_name(request.params["location"])
```

Comment:
In vm and private subnet web routes, we set `@location` in the route. Should we keep it consistent.
https://github.com/ubicloud/ubicloud/blob/190cc6e5706317bd3930398c6702c3d931c2cfa6/routes/merged/project/private_subnet.rb#L11
https://github.com/ubicloud/ubicloud/blob/190cc6e5706317bd3930398c6702c3d931c2cfa6/routes/merged/project/vm.rb#L18